miden-lib/asm/sat/note_setup.masm

```diff
@@ -0,0 +1,38 @@
+use.miden::sat::prologue
+
+# MEMORY OFFSETS
+# =================================================================================================
+
+# BOOK KEEPING
+# -------------------------------------------------------------------------------------------------
+const.CURRENT_CONSUMED_NOTE_PTR=3
+const.CONSUMED_NOTE_INPUTS_HASH_OFFSET=3
+
+# NOTE SETUP SCRIPT
+# =================================================================================================
+
+export.note_setup_script.5
+    # convert the consumed note index to a pointer and store in memory
+    exec.prologue::consumed_note_data_ptr push.CURRENT_CONSUMED_NOTE_PTR mem_store
+
+    # load the note inputs from the advice provider
+    locaddr.0 padw padw padw adv_pipe adv_pipe
```

Comment:
Let's add a `TODO` here to optimize in the future. I think we might be able to structure this so that by the end of the operation we have all inputs on the stack and we don't need to read them back from memory again. But we might have to adjust how `adv_pipe` works a bit.